includes/parser/tests/ParserOutputTest.php

```diff
@@ -0,0 +1,382 @@
+<?php
+
+class ParserOutputTest extends WikiaBaseTest {
+
+	/**
+	 * @dataProvider getParserOutputVars
+	 */
+	public function testMergeExternalParserOutputVars( $sourceVars, $externalVars, $expectedVars ) {
+		$parserOutputSource = new ParserOutput();
+		$parserOutputExternal = new ParserOutput();
+
+		foreach ( $parserOutputSource::$varsToMerge as $var ) {
```

Comment:
Just as a note - I am aware of methods like `addModules`, `addCategory` etc., however mocking a lot of `Title`, `Category` and `Revisions` objects, just to get these arrays would be over-engineering in my opinion. :)